Parser/tokenizer.h

```diff
@@ -43,6 +43,10 @@ enum tokenizer_mode_kind_t {
 typedef struct _tokenizer_mode {
     enum tokenizer_mode_kind_t kind;
 
+    // TODO: we probably can infer this without storing it
+    // from the other information available here.
+    int format_spec;
```

Comment:
We could also add a new tokenizer mode kind for this. I don't know what would simplify this more

Reply:
Was thinking about a mode, but since each mode currently maps to a single get function I thought it wasn't that necessary and this flag can kept in the mode level (and even removed in the future if we can leverage the other meta information here)